spec/lib/krikri/harvesters/oai_harvester_spec.rb

```diff
@@ -329,6 +329,21 @@
                                        .and_return(result)
             subject.send(method, { :skip_set => 'moomin' }).first
           end
+
+          it 'skips sets that error' do
+            args[:oai].delete(:set)
+            invalid = args[:oai].dup
+            invalid[:set] = 'invalid'
+            valid = args[:oai].dup
+            valid[:set] = 'valid'
+
+            allow(subject).to receive(:sets).and_return(['invalid', 'valid', 'moomin'])
+            expect(subject.client).to receive(request_type).with(invalid)
+                                       .and_raise(OAI::Exception, '')
+            expect(subject.client).to receive(request_type).with(valid)
+                                       .and_return(result)
+            subject.send(method, { :skip_set => 'moomin' }).first
+          end
```

Comment:
I guess that I'm not very bothered by the two `expect`s.  If `concat_enum` (in `#request_with_sets`) is tested somewhere than it doesn't seem the place here to test that it works. I just haven't seen it anywhere else and was wondering if the concatenation is verified.  I see references for `collect_concat` / `flat_map`, but not this method. If this test asserts "the errored set is left out but the good one's element is included" then it should be OK.